src/lib/scheduler/abstract_task.cpp

```diff
@@ -81,20 +89,48 @@ void AbstractTask::join() {
 void AbstractTask::_join_without_replacement_worker() {
   std::unique_lock<std::mutex> lock(_done_mutex);
   _done_condition_variable.wait(lock, [&]() { return _done; });
+
+  // An exception was thrown in the task. We caught it there so that the worker does not crash. This might be an
+  // exception from a predecessor task. If you get an exception here, a good idea is to disable the scheduler so
+  // that it throws immediately. Also, you could use gcc with "catch throw"
+  if (_exception) std::rethrow_exception(_exception);
 }
 
 void AbstractTask::execute() {
   DebugAssert(!(_started.exchange(true)), "Possible bug: Trying to execute the same task twice");
   DebugAssert(is_ready(), "Task must not be executed before its dependencies are done");
 
-  _on_execute();
+  try {
+    _on_execute();
+  } catch (...) {
```

Comment:
> * don't catch "serious" issues, only invalid_argument
> * serious issues (e.g., (Debug)Assert should still lead to std::terminate

Reply:
My understanding was, that we want to catch everything here (so that the worker won't die). Then we notify the caller or if there is no worker the throw the exception again. I thought the "don't catch "serious" issues, only invalid_argument" part applied to the specific places where the handle user input

Reply:
No - if we have an actual error, we still want to die as early as possible and not catch them. This is only about exceptions that are caused by user errors (e.g., mismatching data types for insert). *That* is something that the database should survive.

Reply:
Ok, I will change this